checks.d/mongo.py

```diff
@@ -61,6 +58,84 @@ class MongoDb(AgentCheck):
         "metrics.repl.buffer.count",
         "metrics.repl.buffer.maxSizeBytes",
         "metrics.repl.buffer.sizeBytes",
+
+        # mongo 3.x metrics
+        "locks.Collection.acquireCount.R",
+        "locks.Collection.acquireCount.W",
+        "locks.Collection.acquireWaitCount.R",
+        "locks.Collection.acquireWaitCount.W",
+        "locks.Collection.timeAcquiringMicros.R",
+        "locks.Collection.timeAcquiringMicros.W",
+        "locks.Database.acquireCount.r",
+        "locks.Database.acquireCount.w",
+        "locks.Database.acquireWaitCount.r",
+        "locks.Database.acquireWaitCount.w",
+        "locks.Database.timeAcquiringMicros.r",
+        "locks.Database.timeAcquiringMicros.w",
+        "locks.Global.acquireCount.r",
+        "locks.Global.acquireCount.w",
+        "locks.Global.acquireWaitCount.r",
+        "locks.Global.acquireWaitCount.w",
+        "locks.Global.timeAcquiringMicros.r",
+        "locks.Global.timeAcquiringMicros.w",
+        "locks.Metadata.acquireCount.R",
+        "locks.Metadata.acquireCount.W",
+        "locks.MMAPV1Journal.acquireCount.r",
+        "locks.MMAPV1Journal.acquireCount.w",
+        "locks.MMAPV1Journal.acquireWaitCount.r",
+        "locks.MMAPV1Journal.acquireWaitCount.w",
+        "locks.MMAPV1Journal.timeAcquiringMicros.r",
+        "locks.MMAPV1Journal.timeAcquiringMicros.w",
+        "locks.oplog.acquireCount.R",
+        "locks.oplog.acquireCount.w",
+        "locks.oplog.acquireWaitCount.R",
+        "locks.oplog.acquireWaitCount.w",
+        "locks.oplog.timeAcquiringMicros.R",
+        "locks.oplog.timeAcquiringMicros.w",
+
```

Comment:
Yes it can. Datadog Agent takes in account multiple metric type definitions such as `gauge`, `rate`, `monotonic count`, `histogram` etc ... The exhaustive list is available in the [AgentCheck class definition](https://github.com/DataDog/dd-agent/blob/master/checks/__init__.py#L390-L418).
For monotically increasing counters, I'd recommand to submit the metrics with the `rate` metric type. `monotonic_count` would work too but as other counters in the check are reported as `rate`, it's probably better to stick with it. 

In the last version of the check, it should be as easy as adding `RATE` as a value of the metric name in the appropriate metric dictionnary.
c.f. https://github.com/DataDog/dd-agent/blob/master/checks.d/mongo.py#L107